payjoin-defaults/src/lib.rs

```diff
@@ -13,20 +11,49 @@ use payjoin::Url;
 /// parameter is only available when the "danger-local-https" feature is enabled.
 #[cfg(feature = "v2")]
 pub fn fetch_ohttp_keys(
-    ohttp_relay: Url,
-    payjoin_directory: Url,
-    #[cfg(feature = "danger-local-https")] cert_der: Vec<u8>,
+    ohttp_relay: payjoin::Url,
+    payjoin_directory: payjoin::Url,
+) -> Result<payjoin::OhttpKeys, Error> {
+    let ohttp_keys_url = payjoin_directory.join("/ohttp-keys")?;
+    let proxy = reqwest::Proxy::all(ohttp_relay.as_str())?;
+    let client = reqwest::blocking::Client::builder().proxy(proxy).build()?;
+    let mut res = client.get(ohttp_keys_url).send()?;
+    let body = {
+        let mut body = Vec::new();
+        let _ = res.copy_to(&mut body)?;
+        body
+    };
+    payjoin::OhttpKeys::decode(body.as_slice())
+        .map_err(|e| Error(InternalError::InvalidOhttpKeys(e.to_string())))
+}
+
+#[cfg(all(feature = "v2", feature = "danger-local-https"))]
+pub async fn danger_fetch_ohttp_keys(
+    ohttp_relay: payjoin::Url,
+    payjoin_directory: payjoin::Url,
+    cert_der: Vec<u8>,
 ) -> Result<payjoin::OhttpKeys, Error> {
     let ohttp_keys_url = payjoin_directory.join("/ohttp-keys")?;
-    let proxy = PayjoinProxy::new(
-        &ohttp_relay,
-        #[cfg(feature = "danger-local-https")]
-        cert_der,
-    )?;
-    let res = proxy.get(ohttp_keys_url.as_str()).call()?;
-    let mut body = Vec::new();
-    let _ = res.into_reader().read_to_end(&mut body)?;
-    payjoin::OhttpKeys::decode(&body)
+    let proxy = reqwest::Proxy::all(ohttp_relay.as_str())?;
+    let client = reqwest::Client::builder()
+        .danger_accept_invalid_certs(true)
+        .use_rustls_tls()
+        .add_root_certificate(reqwest::tls::Certificate::from_der(cert_der.as_slice()).unwrap())
+        .proxy(proxy)
+        .build()
+        .unwrap();
```

Comment:
handling errors in library code makes it much easier to troubleshoot when errors crop up and is a requirement imo. I see this differently than our own integration tests because other people may well depend on it.